pkg/k8s/api/v1alpha1/telemetryservice_types.go

```diff
@@ -35,11 +36,25 @@ type HTTPSetting struct {
 	Password *string `json:"password,omitempty"`
 }
 
+type MinIOSetting struct {
+	APIId  *string `json:"APIId,omitempty"`
+	APIKey *string `json:"APIKey,omitempty"`
+	//+kubebuilder:validation:Required
+	Bucket *string `json:"Bucket,omitempty"`
+	//+kubebuilder:validation:Required
+	FileExtension *string `json:"FileExtension,omitempty"`
+	//+kubebuilder:validation:Required
+	EndPoint *string `json:"EndPoint,omitempty"`
+	Secret   *string `json:"Secret,omitempty"`
+}
+
 // ServiceSettings defines service settings on telemetry
 type ServiceSettings struct {
-	HTTPSetting *HTTPSetting          `json:"HTTPSetting,omitempty"`
-	MQTTSetting *MQTTSetting          `json:"MQTTSetting,omitempty"`
-	SQLSetting  *SQLConnectionSetting `json:"SQLSetting,omitempty"`
+	RequestTimeout *int64                `json:"RequestTimeout,omitempty"`
```

Comment:
how about create a new field like `connectionSettings` move it into the new field?